birdhouse/config/thredds/docker-compose-extra.yml

```diff
@@ -21,8 +20,8 @@ services:
       PAVICS_FQDN_PUBLIC: $PAVICS_FQDN_PUBLIC
     volumes:
       - thredds_persistence:/usr/local/tomcat/content/thredds
-      - ${DATA_PERSIST_ROOT}/datasets:/pavics-data
-      - ${DATA_PERSIST_ROOT}/ncml:/pavics-ncml
+      - ${MISC_DATA_LOCATION_FOR_THREDDS_ON_HOST}:${MISC_DATA_LOCATION_FOR_THREDDS_ON_CONTAINER}
+      - ${STAC_DATA_LOCATION_FOR_THREDDS_ON_HOST}:${STAC_DATA_LOCATION_FOR_THREDDS_ON_CONTAINER}
```

Comment:
Since it could map to another location (depending on what the user sets), it can potentially cause other locations to be incorrectly mapped because they don't align anymore.

For example: 
https://github.com/bird-house/birdhouse-deploy/blob/2b344d3383d8c0cd3f4cc4a0bf4266e3bae06a47/birdhouse/config/wps_outputs-volume/config/proxy/docker-compose-extra.yml#L6

https://github.com/bird-house/birdhouse-deploy/blob/2b344d3383d8c0cd3f4cc4a0bf4266e3bae06a47/birdhouse/components/weaver/default.env#L85

Considering that `wps_outputs` volume is configured accordingly such that they will appear next to `${DATA_PERSIST_ROOT}/datasets` so that they are all listed under
`https://${HOST}/twitcher/ows/proxy/thredds/catalog/birdhouse/catalog.html`.

Reply:
I think I also agree with you @fmigneault on this one but is there a possible bug with the value of `WEAVER_WPS_OUTPUTS_DIR`?  

If `${DATA_PERSIST_ROOT}/datasets` is `/pavics-data`, then shouldn't `WEAVER_WPS_OUTPUTS_DIR` be `${DATA_PERSIST_ROOT}/datasets/wps_outputs/weaver` (missing `datasets`)?

If it is indeed the case, then `WEAVER_WPS_OUTPUTS_DIR='${MISC_DATA_LOCATION_FOR_THREDDS_ON_HOST}/wps_outputs/weaver'`

Reply:
`WEAVER_WPS_OUTPUTS_DIR='${MISC_DATA_LOCATION_FOR_THREDDS_ON_HOST:-${DATA_PERSIST_ROOT}/datasets}/wps_outputs/weaver'` to avoid Weaver component having direct dependency on Thredds component.

Reply:
The expectation is that the (partial) structure is as follows:
```
/data
    /datasets
    /wps_outputs
        /<bird>
    /magpie_persists
    [...]
```

I can see that there is indeed a `/data/datasets/wps_outputs` that gets created (but empty) due to the `${DATA_PERSIST_ROOT}/datasets:/pavics-data` volume mount. However, I'm not quite sure how the expected `/data/wps_outputs/<bird>` locations get mapped in THREDDS, but they do appear in the right place: 

- https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/catalog/birdhouse/wps_outputs/catalog.html
- https://hirondelle.crim.ca/twitcher/ows/proxy/thredds/catalog/birdhouse/wps_outputs/catalog.html

They are populated with the appropriate outputs from the WPS services.

And https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/catalog/datasets/catalog.html does not have `wps_outputs` somehow?

Reply:
> I'm not quite sure how the expected `/data/wps_outputs/<bird>` locations get mapped in THREDDS, but they do appear in the right place:

Thredds get it from here https://github.com/bird-house/birdhouse-deploy/blob/2b344d3383d8c0cd3f4cc4a0bf4266e3bae06a47/birdhouse/config/thredds/docker-compose-extra.yml#L26

Note the content of folder `wps_outputs/` is from a data-volume `wps_outputs` and not from the disk volume-mount `/data/wps_outputs`.  

That `/data/wps_outputs` path from `WEAVER_WPS_OUTPUTS_DIR` hopefully is the path inside the container, like the birds, and not the path on disk.  If path on disk, it is probably wrong.

We have to be careful when we reference a path, be specific if it is path on the disk of the host or path inside the container.

Each bird also volume-mount the same `wps_outputs` data-volume so that's why everything "works together": https://github.com/bird-house/birdhouse-deploy/blob/2b344d3383d8c0cd3f4cc4a0bf4266e3bae06a47/birdhouse/config/finch/config/wps_outputs-volume/docker-compose-extra.yml#L6


> And https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/catalog/datasets/catalog.html does not have `wps_outputs` somehow?

Because not needed.  We only need one single location for `wps_outputs` on Thredds, for all the WPS services (birds).

Reply:
Just to clarify this situation:

- `wps_outputs` is a volume mounted to each of the birds' containers and to the proxy container at `pavics-data/wps_outputs`
- `${WEAVER_WPS_OUTPUTS_DIR}` is a location on host that is mounted to the weaver, weaver-worker, and proxy container at `pavics-data/wps_outputs/weaver`

`${WEAVER_WPS_OUTPUTS_DIR}` needs to be a bind mount, not a volume, because it needs to be mounted in the container at the same location as on the host so that it can be mounted to the weaver-worker containers by a process running in the weaver container.

@fmigneault @tlvu is that correct?

Reply:
I'll let @fmigneault reply for Weaver, I am not fully familiar with it.

Reply:
I've now incorporated all the other comments. Any thoughts on what we need to do about this?

Reply:
As long as the mapping paths are the same on both sides of `:` for Weaver, it doesn't matter where the directory is actually located. 

Other than that, the paths are only "conveniently" named in such a way that they match `<data-root>/wps_outputs/<bird>` format to align with other services that do not have this "same path" requirement as Weaver does. All those paths need to align to allow Cowbird to perform permission synchronization between WPS-outputs file access, THREDDS, Jupyter workspaces, etc.

If all those conditions are met, there shouldn't be any issue with how paths/env-vars are used to define mounts. 
However, I would like to avoid the possibility for inexperienced users that are not aware of all these conditions and implications to inadvertently break functionalities simply because they changed some base directory path.

Reply:
What is the right `wps_output` dir? Is it: (a) `${DATA_PERSIST_ROOT}/wps_outputs` or (b) `${DATA_PERSIST_ROOT}/datasets/wps_outputs`? If it is (a), and from the responses it sounds like it is (a), then there shouldn't be an issue with the change proposed because they don't affect the folders `${DATA_PERSIST_ROOT}/wps_outputs` and `${DATA_PERSIST_ROOT}/user_workspaces`.

Reply:
Yes, I believe it should be (a).